File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/controller/book/RegisterBookController.java
package dev.starryeye.book_library.controller.book;

import dev.starryeye.book_library.application.book.command.RegisterBookService;
import dev.starryeye.book_library.controller.book.request.RegisterBookRequest;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class RegisterBookController {

    private final RegisterBookService service;

    @PostMapping("/api/v1/books/new")
    public void registerBook(@Valid @RequestBody RegisterBookRequest request) {
        service.register(request.toCommand());
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/book/RegisterBookController.kt
package dev.starryeye.book_library.controller.book

import dev.starryeye.book_library.application.book.command.RegisterBookService
import dev.starryeye.book_library.controller.book.request.RegisterBookRequest
import dev.starryeye.book_library.controller.book.request.toCommand
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class RegisterBookController(
    private val service: RegisterBookService,
) {

    @PostMapping("/api/v1/books/new")
    fun bookRegister(
        @Valid @RequestBody request: RegisterBookRequest,
    ) {
        service.register(request.toCommand())
    }
}